pkg/querier/http.go

```diff
@@ -82,13 +84,50 @@ func (q *QueryHandlers) Render(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	res, err := q.upstream.SelectMergeStacktraces(req.Context(), connect.NewRequest(selectParams))
+
+	var resFlame *connect.Response[querierv1.SelectMergeStacktracesResponse]
+	g, ctx := errgroup.WithContext(req.Context())
+	g.Go(func() error {
+		resFlame, err = q.upstream.SelectMergeStacktraces(ctx, connect.NewRequest(selectParams))
+		return err
+	})
+
+	timelineStep := timeline.CalcPointInterval(selectParams.Start, selectParams.End)
+	var resSeries *connect.Response[querierv1.SelectSeriesResponse]
+	g.Go(func() error {
+		resSeries, err = q.upstream.SelectSeries(req.Context(),
+			connect.NewRequest(&querierv1.SelectSeriesRequest{
+				ProfileTypeID: selectParams.ProfileTypeID,
+				LabelSelector: selectParams.LabelSelector,
+				Start:         selectParams.Start,
+				End:           selectParams.End,
+				Step:          timelineStep,
+			}))
+
+		return err
+	})
+
+	err = g.Wait()
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
+	if len(resSeries.Msg.Series) > 1 {
+		http.Error(w, fmt.Sprintf("can't construct timeline since multiple series have been returned: '%d'", len(resSeries.Msg.Series)), http.StatusBadRequest)
```

Comment:
Should never be the case since you're not passing a by parameters btw.

Reply:
good point